src/core/fetch/proposal.ts

```diff
@@ -0,0 +1,109 @@
+import { Proposal } from '../types/Proposal';
+import { genericFetchAndThrowIfError } from './common/generic';
+import { routes } from './common/routes';
+import {
+  GetAllProposalsParams,
+  CreateProposalParams,
+  UpdateProposalParams,
+  ProposalParams,
+} from './common/params';
+
+/**
+ * Fetches all proposals for a specific DAO.
+ * @param params - The parameters for fetching proposals.
+ * @param params.chainId - The ID of the chain.
+ * @param params.address - The address of the DAO.
+ * @param params.apiUrl - Optional API URL override.
+ * @returns A promise that resolves to an array of proposals.
+ */
+export const getAllProposals = async (params: GetAllProposalsParams): Promise<Proposal[]> => {
+  const { chainId, address, apiUrl } = params;
+  const proposals = await genericFetchAndThrowIfError<Proposal[]>({
+    route: routes.proposal(chainId, address),
+    apiUrl,
+  });
+  return proposals;
+};
+
+/**
+ * Fetches a specific proposal by its slug.
+ * @param params - The parameters for fetching a proposal.
+ * @param params.chainId - The ID of the chain.
+ * @param params.address - The address of the DAO.
+ * @param params.slug - The slug of the proposal.
+ * @param params.apiUrl - Optional API URL override.
+ * @returns A promise that resolves to the proposal.
+ */
+export const getProposal = async (params: ProposalParams): Promise<Proposal> => {
+  const { chainId, address, slug, apiUrl } = params;
+  const proposal = await genericFetchAndThrowIfError<Proposal>({
+    route: `${routes.proposal(chainId, address)}/${slug}`,
```

Comment:
```typescript
// https://github.com/decentdao/decent-app/blob/6c8b58ab330a37098828e8e8f8b3c8dc14a3c92b/src/constants/routes.ts#L18-L21
hierarchy: {
    relative: (addressPrefix: string, safeAddress: string) =>
      `/hierarchy${getDaoQueryParam(addressPrefix, safeAddress)}`,
    path: 'hierarchy',
  },
```
maybe we can extract all possible route into `common/routes`, check [example](https://github.com/decentdao/decent-app/blob/6c8b58ab330a37098828e8e8f8b3c8dc14a3c92b/src/constants/routes.ts#L18-L21) above.
the idea is we shouldn't put `/` here to construct route anymore, all routes are returned by invoking function with args.

Reply:
oooo I see now this is nice lets do this